notifiers/datadog/datadog.go

```diff
@@ -0,0 +1,173 @@
+package notifiers
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/zorkian/go-datadog-api"
+	"strings"
+
+	dependencies "github.com/InVisionApp/kit-overwatch/deps"
+	"github.com/InVisionApp/kit-overwatch/notifiers/deps"
+)
+
+const EVENT_TYPE = "kubernetes"
+const MD_PREFIX = "%%% \n"
+const MD_SUFFIX = "\n %%%"
+
+type NotifyDataDog struct {
+	Client dependencies.IDataDogClient
+}
+
+func New(client dependencies.IDataDogClient) *NotifyDataDog {
+	return &NotifyDataDog{
+		Client: client,
+	}
+}
+
+func (ndd *NotifyDataDog) Send(n *deps.Notification) error {
+
+	if n == nil {
+		return fmt.Errorf("Notification cannot be nil")
+	}
+
+	event := &datadog.Event{
+		SourceType: EVENT_TYPE,
+		EventType:  EVENT_TYPE,
+	}
+
+	switch n.Level {
+	case "INFO":
+		event.AlertType = "Info"
+		event.Priority = "low"
+	case "WARN":
+		event.AlertType = "Warning"
+		event.Priority = "high"
+	case "ERROR":
+		event.AlertType = "Error"
+		event.Priority = "high"
+	}
+
+	serviceName := n.Event.ObjectMeta.Name
+	splitName := strings.Split(n.Event.ObjectMeta.Name, "-")
+
+	switch n.Event.InvolvedObject.Kind {
+	case "Pod":
+		// Default Rules to ignore `service-name-[deploynumber-podnumber.number]`
```

Comment:
even though this is only a comment, technically `service-name-[deploynumber-podnumber.number]` is really the `service-name-[replicaset-podnumber.number]`